### DISCUSSION TITLE
Healing Summary

### DISCUSSION THREAD

**Discussion by moljor** [minio/minio#14657] [Q&A]
Dear,

After a node replacement all disks in the healing status show "OK" instead of "HEALING" but the healing summary shows for example:

Summary:
=======
Objects Healed: 137/455 (30.1%), 336 kB/854 kB (39.3%)
Heal rate: 0 obj/s, 53 B/s
Estimated Completion: 4 hours

Is the healing still actively doing something?
It's been stuck like that for quite a while and running an mc stat on my buckets seemingly shows all items.

Is there a way to verify if healing is completely done?

Thanks!

**Comment by harshavardhana** [comment]
Various types of healing is always running to fix inconsistencies

**Reply by moljor** [reply]
Ok but say that I need to upgrade all disks in a cluster, node per node. When do I know it's safe to proceed to the next node, ultimately replacing all disks?

**Reply by harshavardhana** [reply]
> Ok but say that I need to upgrade all disks in a cluster, node per node. When do I know it's safe to proceed to the next node, ultimately replacing all disks?

You can replace disks as long as you do not replace more than the parity number of disks in the cluster.

**Reply by moljor** [reply]
So if you replace less than the parity number of disks, when do you know it's safe to replace the rest? Is it possible to know this?

**Reply by networkjackbb** [reply]
From what I've seen, after replacing a certain set of drives, in order to know that it is safe to replace a subsequent set of disks, you would need to go bucket by bucket and check the Health Status. I know that you can do that in the Console under Monitoring/Drives. and you can see the After Healing/Before Healing percentages. If they are both 100%, then you know that bucket is "safe". Do that for all your buckets and the cluster is good. It'd be nice if there were a better way to manage this, but if there is, I haven't been able to locate it.
